File: Arrays & Hashing/code/replace_elements.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
        int n = arr.size();
        vector<int> res(n);
        int maxNum = arr[n - 1];

        for(int i = n - 2; i >= 0; i--) {
            maxNum = max(maxNum, arr[i + 1]);
            res[i] = maxNum;
        }
        res[n - 1] = -1;

        return res;
    }
};

File: Arrays & Hashing/code/replace_elements.go
func replaceElements(arr []int) []int {
	size := len(arr)
	res := make([]int, size)
	num := math.MinInt
	
	for right := size - 2; right >=0 ; right-- {
		num = maxNum(num, arr[right + 1])
		res[right] = num
	}

	res[size - 1] = -1

	return res
}

func maxNum (a, b int) int {
	if a > b {
		return a
	}
	return b
}
